include/inexor/vulkan-renderer/world/cube.hpp

```diff
@@ -102,19 +104,47 @@ class Cube : public std::enable_shared_from_this<Cube> {
 
     /// Is the current cube root.
     [[nodiscard]] bool is_root() const noexcept;
+    /// Is the current cube a leaf.
+    [[nodiscard]] bool is_leaf() const noexcept;
     /// At which child level this cube is.
     /// root cube = 0
     [[nodiscard]] std::size_t grid_level() const noexcept;
     /// Count the number of Type::SOLID and Type::NORMAL cubes.
     [[nodiscard]] std::size_t count_geometry_cubes() const noexcept;
 
+    [[nodiscard]] glm::vec3 center() const noexcept {
+        return m_position + 0.5f * m_size;
+    }
+
+    [[nodiscard]] const glm::vec3 &position() const noexcept {
+        return m_position;
+    }
+
+    [[nodiscard]] std::array<glm::vec3, 2> bounding_box() const {
+        return {m_position, {m_position.x + m_size, m_position.y + m_size, m_position.z + m_size}};
+    }
+
+    [[nodiscard]] float bounding_box_radius() const {
+        return static_cast<float>(std::sqrt(3) * m_size) / 2.0f;
+    }
+
+    [[nodiscard]] float size() const noexcept {
+        return m_size;
+    }
+
+    /// @brief Calculate the square of the distance from the center of the cube to a given position.
+    /// @param pos The position of which the squared distance to the center of the cube will be calculated.
+    [[nodiscard]] float squared_distance(const glm::vec3 pos) const {
```

Comment:
Not sure if this should be a cube method.

Reply:
I will move this into the `RayCubeCollision` class, where we can use this macro:

```cpp
const auto square_of_distance = [&](const std::array<glm::vec3, 2> &points) {
    return glm::distance2(points[0], points[1]);
};
```

This will also reduce code duplication.

Reply:
Done.